linkedin.py

```diff
@@ -161,74 +161,92 @@ def processJob(self, jobID: str, jobCounter: models.JobCounter):
-                    if config.displayWarnings:
-                        prYellow(f"⚠️  Not adding a job as the company '{companyName}' is blacklisted")
-                    continue
-
-            if companyName is None:
-                utils.logDebugMessage("Couldn't find companyName", utils.MessageTypes.WARNING)
+            jobTitle = self.getJobTitleFromJobCard(jobItem)
+            if not jobTitle:
+                utils.logDebugMessage("Could not extract job title from job card", MessageTypes.WARNING)
                 continue
 
-            jobTitleAnchor = jobItem.find_elements(By.XPATH, ".//a[contains(@class, 'job-card-container__link job-card-list__title')]")
-            if len(jobTitleAnchor) > 0:
-                allTexts = jobTitleAnchor[0].text.split("\n")
-                uniqueTexts = list(dict.fromkeys(allTexts))
-                jobTitle = uniqueTexts[0].strip()
+            if self.isTitleBlacklisted(jobTitle):
+                utils.logDebugMessage(f"Not adding job as title '{jobTitle}' is blacklisted", MessageTypes.INFO)
+                continue
 
-                if self.isTitleBlacklisted(jobTitle):
-                    if config.displayWarnings:
-                        prYellow(f"⚠️  Not adding a job as the title '{jobTitle}' is blacklisted")
-                    continue
-            
-            if jobTitle is None:
-                utils.logDebugMessage("Couldn't find jobTitle", utils.MessageTypes.WARNING)
+            companyName = self.getCompanyNameFromJobCard(jobItem)
+            if not companyName:
+                utils.logDebugMessage("Could not extract company name from job card", MessageTypes.WARNING)
                 continue
-            
-            if jobTitle is None:
-                utils.logDebugMessage("Couldn't find jobTitle", utils.MessageTypes.WARNING)
+
+            if self.isCompanyBlacklisted(companyName):
+                utils.logDebugMessage(f"Not adding job as company '{companyName}' is blacklisted", MessageTypes.INFO)
                 continue
-            
-            jobId = jobItem.get_attribute("data-occludable-job-id")
 
-            if jobId is None:
-                utils.logDebugMessage("Couldn't find jobID", utils.MessageTypes.WARNING)
+            jobId = jobItem.get_attribute("data-occludable-job-id")
+            if not jobId:
+                utils.logDebugMessage("Could not extract job ID from job card", MessageTypes.WARNING)
                 continue
 
+            workPlaceType = self.getWorkplaceTypeFromJobCard(jobItem)
+
             jobsForVerification.append(models.JobForVerification(
                 linkedinJobId=jobId.split(":")[-1],
                 title=jobTitle,
                 company=companyName,
                 workplaceType=workPlaceType))
 
         return jobsForVerification
-    
+
+
+    def getCompanyNameFromJobCard(self, jobItem) -> Optional[str]:
+        """Extract company name from a job card using LinkedIn's current structure"""
```

Comment:
Unnecessary comment as the function name is descriptive enough. Although the function name could be more explicit in terms of where is this jobCard, e.g JobCardInSearchResults

Linkedin current structure keywords are not as helpful as the file is called linkedin.py